metric_providers/powermetrics/provider.py

```diff
@@ -35,43 +41,44 @@ def __init__(self, resolution, skip_check=False):
             '-o',
             self._filename]
 
+
     def check_system(self):
-            if self.is_powermetrics_running():
-                raise MetricProviderConfigurationError('Another instance of powermetrics is already running on the system!\nPlease close it before running the Green Metrics Tool.')
+        if self._pm_process_count > 0:
+            raise MetricProviderConfigurationError('Another instance of powermetrics is already running on the system!\n'
+                                                   'Please close it before running the Green Metrics Tool.\n'
+                                                   'You can also override this with --skip-system-checks\n')
 
-    def is_powermetrics_running(self):
-        ps = subprocess.run(['pgrep', '-qx', 'powermetrics'], check=False)
-        if ps.returncode == 1:
-            return False
-        return True
+    def is_our_powermetrics_running(self):
+        pw_processes = subprocess.check_output(['pgrep', '-ix', 'powermetrics'], text=True)
+        total_count = len(pw_processes.strip().split('\n')) if pw_processes else 0
+        minus_startup = total_count -  self._pm_process_count
+        return  minus_startup >= 1
 
 
     def stop_profiling(self):
         try:
             # We try calling the parent method but if this doesn't work we use the more hardcore approach
             super().stop_profiling()
         except PermissionError:
-            #This isn't the nicest way of doing this but there isn't really any other way that is nicer
+            # This isn't the nicest way of doing this but there isn't really any other way that is nicer that doesn't
```

Comment:
I would like to have proper explanatory comment here, because we will forget why we implemented it that way.

Also the comment in the "try" block before is very misleading.

Suggestion:
```
try:
            # We try calling the parent method but since we do not have sudo rights the process running with root rights will ignore the signal. We still call the method because it might contain tear-down functionality in the future that we want to trigger
            super().stop_profiling()
        except PermissionError:
            # We will land here in any case as stated before (root permissions missing). When we trigger *killall* now the process will be terminated. We opted for this implementation as other processes on the system, like for instance the power hog (https://github.com/green-coding-berlin/hog) should not be affected too much. They restart the process anyway when it gets killed. However manual processes that the user might have started will also be killed, so we issue a notice.
# There is really no better way of doing this as of now. Keeping the process id for instance in a hash and killing only that would also fail du to root permissions missing. If we add an /etc/sudoers entry with a wildcard for a PID we open up a security hole. Happy to take suggestions on this one!

```